Update insert_sic feature

apcraig opened the pull request:
Update insert_sic feature
- Rename insert_sic to restart_mod, char string
- Add 'none', 'adjust_aice', and 'adjust_aice_test' restart_mod options
- Update documentation, add data assimilation section
- Add adjust_aice_test testcase

apcraig committed:
Update insert_sic feature

- Rename insert_sic to restart_mod, char string
- Add 'none', 'adjust_aice', and 'adjust_aice_test' restart_mod options
- Update documentation, add data assimilation section
- Add adjust_aice_test testcase

sanAkel commented on cicecore/cicedyn/infrastructure/ice_restart_driver.F90:
```diff
@@ -1232,9 +1248,15 @@ subroutine direct_insert_sic
+        if (present(test)) then
+           ltest = test
+        endif
+        aice_filename = trim(restart_dir)//'sic.nc'
```
Would you please add `/` to the `aice_filename`, like:
```
aice_filename = trim(restart_dir)//'/sic.nc'
```

sanAkel reviewed (changes requested):
@apcraig 

Following tests passed:
1. It built.
2. Ran:
- With default, i.e., `restart_mod`: `none`.
- Test: `adjust_aice_test`
- _Real_ case: `adjust_aice`:

Please add a forward slash:
```
aice_filename = trim(restart_dir)//'/sic.nc'
```

Without it (i.e., your code as-is) kept looking for a file name with some characters from buffer included, ⬆️ modified code ran fine; see below.
```
160:  (ice_check_nc) No such file or directory, (ice_open_nc) ERROR: Cannot open ./RE
160:  START/sic.nc^A^@^@^@^A^@^@^@^A^@^@^@^A^@^@^@^^^B^@^@^@^@^@^@ð^A^@^@^@^@^@^@ ^Z^D^@^@^@^@^@¸¦uÀü^?^@^@^@::üÕ^T^@^@^D^@^@^@^@^@^@^@^@^@^@
160:  ^@^@^@^@^@<81>^@^@^@^@^@^@^@^B^@^@^@^@^@^@^@<98><95><86><99>Ô^T^@^@^W^@^@^@^@^@^@^@^D^@^@^@^@^@^@^@^A^@^@^@^@^@^@^@^B^@^@^@^@^@^@^@\^@^@^@^@^@^@^@^A^@^@^@^@^@^@^@<98>v
160:  Ø<9f>Ô^T^@^@^H^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@<81>^@^@^@^@^@^@^@^A^@^@^@^@^@^@^@<98><z<99>Ô^T^@^@^E^@^@^@^@^@^@^@ ^Z^D^@^@^@^@^@^A^@^@^@^@^@^@^@Ø^F«<9f>Ô^T^@^@^H
160:  ^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@
```

With the added `/` in the file name, I got:
```
(direct_adjust_aice) direct_adjust_aice from ./RESTART//sic.nc
(ice_read_nc_xy) fid=    65536, lnrec =        1, varname = sic
```

apcraig committed:
Update sic.nc filename format

Update rounding to 5/100th from 1/100th.  Tested with the standalone gx3 case.
One thing that is not tested with the adjust_aice_test option is adding
sea ice where there was none originally.

sanAkel reviewed (approved):
Thanks @apcraig
